classes/SubjectDemographics.php

```diff
@@ -89,54 +88,51 @@ public function setAllSubjects(): void
 
 
     /**
+     * @param Guzzle\Client $client
      * @param $mrn
      * @param $subjectSource
      * @return array|mixed|void
      * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function getOnCoreSubjectDemographics($subjectDemographicsId, $forceDemographicsPull = false)
-    {
-        try {
-            // check if entity table already has that subject otherwise go to API to get info
-            $demo = $this->getAllSubjects()[$subjectDemographicsId];
-            if (empty($demo) || $forceDemographicsPull) {
-                $response = $this->getUser()->get('subjectDemographics/' . $subjectDemographicsId);
-
-                if ($response->getStatusCode() < 300) {
-                    $data = json_decode($response->getBody(), true);
-                    if (empty($data)) {
-                        return [];
-                    } else {
-                        // create entity table record before return.
-                        #$entity = (new Entities)->getFactory()->create(OnCoreIntegration::ONCORE_SUBJECTS, $data);
-                        if($demo){
-                            $entity = (new Entities)->update(OnCoreIntegration::ONCORE_SUBJECTS, $demo['id'], $data);
-                        }
-                        else{
-                            $entity = (new Entities)->create(OnCoreIntegration::ONCORE_SUBJECTS, $data);
-                        }
-                        if ($entity) {
-                            return $data;
-                        } else {
-                            Entities::createLog('cant create subject record for ' . $data['subjectDemographicsId']);
-                            Entities::createLog(implode('<br>', $data));
-                        }
-
-                    }
+    public function getOnCoreSubjectDemographicsAsync($client, $subjectDemographicsId, $forceDemographicsPull = false)
+    {
+        return new Promise\Promise(function () use ($client, $subjectDemographicsId, $forceDemographicsPull, &$promise) {
+            try {
+                $demo = $this->getAllSubjects()[$subjectDemographicsId] ?? null;
+                if (empty($demo) || $forceDemographicsPull) {
+
+                    $client->getAsync('subjectDemographics/' . $subjectDemographicsId)
```

Comment:
getAsync() already returns a promise, do you need to wrap everything in a new Promise ?

Reply:
yes, because only getAsync returns the promise rest of the code does not. so I needed an extra check to make sure everything was synced correctly.